backend/src/main/resources/db/migration/V1.10__pipeline_version_per_organism.sql

```diff
@@ -0,0 +1,116 @@
+    case
+        when all_external_metadata.external_metadata is null then
+            jsonb_build_object('metadata', (cpd.processed_data->'metadata'))
+        else
+            jsonb_build_object(
+                'metadata',
+                (cpd.processed_data->'metadata') || all_external_metadata.external_metadata
+            )
+    end as joint_metadata
+from
+    (
+        -- Get only the preprocessed data for the current pipeline version
+        select * from sequence_entries_preprocessed_data
+        where pipeline_version = (select version from current_processing_pipeline
+                                  where organism = (
+                                      select organism from sequence_entries se
+                                      where se.accession = sequence_entries_preprocessed_data.accession
+                                      and se.version = sequence_entries_preprocessed_data.version
+                                  ))
+    ) cpd
+    left join all_external_metadata on
+        all_external_metadata.accession = cpd.accession
+        and all_external_metadata.version = cpd.version;
+
+drop view if exists sequence_entries_view;
+
+create view sequence_entries_view as
+select
+    se.*,
+    sepd.started_processing_at,
+    sepd.finished_processing_at,
+    sepd.processed_data as processed_data,
+    sepd.processed_data || em.joint_metadata as joint_metadata,
+    case
+        when se.is_revocation then (select version from current_processing_pipeline
+                                    where organism = se.organism)
+        else sepd.pipeline_version
+    end as pipeline_version,
+    sepd.errors,
+    sepd.warnings,
+    case
+        when se.released_at is not null then 'APPROVED_FOR_RELEASE'
+        when se.is_revocation then 'PROCESSED'
+        when sepd.processing_status = 'IN_PROCESSING' then 'IN_PROCESSING'
+        when sepd.processing_status = 'PROCESSED' then 'PROCESSED'
+        else 'RECEIVED'
+    end as status,
+    case
+        when sepd.processing_status = 'IN_PROCESSING' then null
+        when sepd.errors is not null and jsonb_array_length(sepd.errors) > 0 then 'HAS_ERRORS'
+        when sepd.warnings is not null and jsonb_array_length(sepd.warnings) > 0 then 'HAS_WARNINGS'
+        else 'NO_ISSUES'
+    end as processing_result
+from
+    sequence_entries se
+    left join sequence_entries_preprocessed_data sepd on
+        se.accession = sepd.accession
+        and se.version = sepd.version
+        and sepd.pipeline_version = (select version from current_processing_pipeline
+                                     where organism = (
+                                         select organism from sequence_entries se
+                                         where se.accession = sepd.accession
+                                         and se.version = sepd.version
+                                     ))
```

Comment:
Same comment as above: I think that it would be more readable to just join the tables instead of using nested queries. (But this is also correct, so not blocking)

Reply:
https://github.com/loculus-project/loculus/pull/3534/commits/9c8e7cac8bc748c5a2c544339d0d522262f66d8c